apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPreparedStatementSetterInterceptor.java

```diff
@@ -24,13 +24,14 @@
 import org.apache.skywalking.apm.plugin.jdbc.define.StatementEnhanceInfos;
 
 import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 
 public class JDBCPreparedStatementSetterInterceptor implements InstanceMethodsAroundInterceptor {
     @Override
     public final void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
         Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
         final StatementEnhanceInfos statementEnhanceInfos = (StatementEnhanceInfos) objInst.getSkyWalkingDynamicField();
-        if (statementEnhanceInfos != null) {
+        if (statementEnhanceInfos != null && Modifier.isPublic(method.getModifiers())) {
```

Comment:
If you need to change the intercepting points, please change the codes of matcher in the instrument class.

Reply:
Do you mean don't change the interceptor code? 

In general, non-public method is the underlying implementation, I think it need not to be intercept.
So if I can change the PSSetterDefinitionOfJDBCInstrumentation code,like this:
```
@Override
    public ElementMatcher<MethodDescription> getMethodsMatcher() {
        ElementMatcher.Junction<MethodDescription> matcher = none();

        // remove TRACE_SQL_PARAMETERS judgement for dynamic config
        final Set<String> setters = ignorable ? PS_IGNORABLE_SETTERS : PS_SETTERS;
        for (String setter : setters) {
            matcher = matcher.or(named(setter));
        }

        return matcher;
    }
```
change `matcher = matcher.or(named(setter))` to `matcher = matcher.or(named(setter).and(isPublic()))`

Reply:
or just change the mssql plugin's Instrumentation.  I only find this problem in the mssql's interface.

Reply:
According to the error, it seems mssql JDBC driver has some special public method, which isn't expected to trace.
As a result, it ends in ClassCastException.
Is this the case?

Changing mssql plugins only work for me, and adding `isPublic` to filter unnecessary intercepting points make sense too if there is no side-effect for other drivers.

Reply:
Yes, it's ClassCastException. For detail:
`java.lang.ClassCastException: com.microsoft.sqlserver.jdbc.Parameter cannot be cast to java.lang.Integer`

It happens in this piece of code:
```
final void setObject(Parameter param, Object obj, JavaType javaType, JDBCType jdbcType, Integer scale,
            Integer precision, boolean forceEncrypt, int parameterIndex, String tvpName) throws SQLServerException 
```
a non-public method

Normally, java.sql.PreparedStatement, The first param type is int of this interface's setXX method. But mssql's underlying implementation is non-standard.

And JDBCPreparedStatementSetterInterceptor intercepts methods like:
"setDouble", "setFloat", "setInt", "setLong", "setNString", "setObject", "setAsciiStream", "setBinaryStream", "setBlob", "setBytes" ...

This methods just set params for preparedstatement. SW just intercept public methods，not non-public methods.

Reply:
You could try to set it for public only, I think. As we should only care about those methods.